tests/function/test_function_return_values_match_reference.py

```diff
@@ -0,0 +1,156 @@
+            """ `test_function()` match the reference value(s)."""
+        ),
+    },
+    {  # Type difference in returned values
+        "submission": "def test_function():\n    return 1234",
+        "reference": "def test_function():\n    return '1234'",
+        "result": AssertionError,
+        "score": 0,
+        "options": Options(
+            obj_name="test_function",
+            sub_module="submission",
+            ref_module="reference",
+            weight=1,
+        ),
+        "message": "Double check the type of the value(s) returned",
+        "doc_func_test_string": (
+            """Check that the value(s) returned from your"""
+            """ `submission.test_function` function when called as """
+            """`test_function()` match the reference value(s)."""
+        ),
+    },
+    {  # Value difference in returned values
+        "submission": "def test_function():\n    return 12345",
+        "reference": "def test_function():\n    return 1234",
+        "result": AssertionError,
+        "score": 0,
+        "options": Options(
+            obj_name="test_function",
+            sub_module="submission",
+            ref_module="reference",
+            weight=1,
+        ),
+        "message": "Double check the value(s) returned",
+        "doc_func_test_string": (
+            """Check that the value(s) returned from your"""
+            """ `submission.test_function` function when called as """
+            """`test_function()` match the reference value(s)."""
+        ),
+    },
+]
+
+
+@pytest.fixture(params=cases)
+def case_test_method(request, fix_syspath):
+    """Arrange submission directory, and parameterized test function."""
+    case = request.param
+    file_path = fix_syspath / f"{case['options'].sub_module}.py"
+    file_path.write_text(case["submission"])
+    file_path = fix_syspath / f"{case['options'].ref_module}.py"
+    file_path.write_text(case["reference"])
+
+    built_class = build(case["options"])
+    built_instance = built_class(
+        methodName="test_function_return_values_match_reference_0"
+    )
+    test_method = built_instance.test_function_return_values_match_reference_0
+
+    return case, test_method
+
+
+def test_function_return_values_match_reference(case_test_method):
+    """Test response of test_submitted_files function."""
+    case, test_method = case_test_method
+
+    if case["result"] == "pass":
+        test_method()  # should not raise an error
+        assert test_method.__score__ == case["score"]
+        assert test_method.__doc__ == case["doc_func_test_string"]
+
+    else:
+        error = case["result"]
+        with pytest.raises(error) as exc_info:
+            test_method()
+        message = " ".join(str(exc_info.value).split())
+        assert case["message"] in message
+        assert test_method.__doc__ == case["doc_func_test_string"]
+        assert test_method.__score__ == case["score"]
```

Comment:
Likewise this could be simplified to checking if the score is equal to zero since the test would have failed.